### PR TITLE
update json encoding from iso-8859-1 to utf8 to allow unicode characters

### PR DESCRIPTION
**Pull request by bkhattak**
Summary
This PR fixes JSON request body encoding in ARM requests by using UTF-8 instead of ISO-8859-1.

This resolves issues with non-ASCII payloads, where Unicode values could be encoded incorrectly or rejected by the API.

Changes
* Send JSON request bodies as UTF-8 in ARM requests
* Add a regression test covering Unicode request payloads
* Consolidate HTTP request test helpers so request-capture tests use a consistent MockHttpMessageHandler(

### CODE REVIEW

File: src/Payload/Payload.csproj [hunk 1]
```diff
@@ -4,7 +4,7 @@
     <RootNamespace>Payload</RootNamespace>
     <TargetFrameworks>netstandard2.0;net461</TargetFrameworks>
     <AssemblyTitle>Payload</AssemblyTitle>
-    <VersionPrefix>2.2.1</VersionPrefix>
+    <VersionPrefix>2.2.2</VersionPrefix>
     <Version>2.2.1</Version>
```
Comment (ianhalpern) [inline src/Payload/Payload.csproj]: We also need to bump this to 2.2.2

**Review by ianhalpern — CHANGES_REQUESTED** [review]
We need to update the version in csproj to 2.2.3

**Review by ianhalpern — APPROVED** [review]
